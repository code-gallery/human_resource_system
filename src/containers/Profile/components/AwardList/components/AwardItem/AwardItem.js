import React from 'react'
import PropTypes from 'prop-types'
import AwardItemView from '../AwardItemView'
import AwardItemEdit from '../AwardItemEdit'

class AwardItem extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      edit: false
    }
  }

  componentWillReceiveProps = (nextProps) => {
    if (this.state.edit && this.props.editMode && !nextProps.editMode) {
      this.toggleEdit()
    }
  }

  toggleEdit = () => {
    this.setState({
      edit: !this.state.edit
    })
    this.props.toggleBlockEditing()
  }

  render() {
    const { edit } = this.state
    const { deleteAward } = this.props
    return (
      <div>
        <AwardItemView
          {...this.props}
          toggleEdit={this.toggleEdit}
        />
        {
          edit &&
          <AwardItemEdit
            {...this.props}
            toggleEdit={this.toggleEdit}
            deleteAward={deleteAward}
            edit={true}
          />
        }
      </div>
    )
  }
}

AwardItem.propTypes = {
  editMode: PropTypes.bool,
  toggleBlockEditing: PropTypes.func.isRequired,
  deleteAward: PropTypes.func.isRequired,
  saveAward: PropTypes.func.isRequired
}

export default AwardItem
