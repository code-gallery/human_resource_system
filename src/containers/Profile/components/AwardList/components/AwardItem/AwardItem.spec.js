import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import AwardItem from './AwardItem'

describe('containers/Profile/components/.../AwardItem', () => {
  const commonProps = {
    name: 'Award name',
    org: {
      id: 5,
      name: 'AA plc',
      logo_image: 'https://appii.s3.eu-west-2.amazonaws.com/organisation_logos/aa-plc1509336690734.jpg'
    },
    created_at: '2017-12-07 15:36:20',
    verified_status: 'not_verified',
    type: 'award',
    visibility: 'registered',
    toggleBlockEditing: jest.fn(),
    deleteAward: jest.fn(),
    saveAward: jest.fn()
  }

  describe('when editMode is false', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps,
        editMode: false
      }
      const tree = renderer.create(
        <AwardItem {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when editMode is true', () => {
    const props = {
      ...commonProps,
      editMode: true
    }

    it('renders correctly', () => {
      const tree = renderer.create(
        <AwardItem {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })

    describe('componentWillReceiveProps', () => {
      it('calls props.toggleBlockEditing correctly', () => {
        const wrapper = shallow(
          <AwardItem {...props} />
        )
        expect(props.toggleBlockEditing).not.toHaveBeenCalled()
        wrapper.setState({ edit: true }, function() {
          wrapper.setProps({
            editMode: false
          })
        })
        expect(props.toggleBlockEditing).toHaveBeenCalled()
        expect(wrapper.state('edit')).toEqual(false)
      })
    })
  })
})
