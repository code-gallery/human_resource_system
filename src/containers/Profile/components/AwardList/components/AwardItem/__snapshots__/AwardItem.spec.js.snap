// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/.../AwardItem when editMode is false renders correctly 1`] = `
<div>
  <div
    className="profile-experience-block profile-unverified-status"
  >
    <div>
      <div
        className="relative"
      >
        <div
          className="Avatar Avatar-white Avatar-size-109"
        >
          <img
            alt="logo"
            src="https://appii.s3.eu-west-2.amazonaws.com/organisation_logos/aa-plc1509336690734.jpg"
          />
        </div>
        <p
          className="visible-xs visible-sm"
        >
          <i
            className="profile-status-ico tooltip-position-top tooltip-movable tooltip-dark"
            data-tooltip="Not verified"
          >
             
          </i>
        </p>
      </div>
      <div
        className="profile-experience-info"
      >
        <i
          className="profile-status-ico tooltip-position-top tooltip-movable tooltip-dark hidden-xs hidden-sm"
          data-tooltip="Not verified"
        >
           
        </i>
        <p>
          Award name
        </p>
        <span>
          AA plc
        </span>
        <em>
          
        </em>
      </div>
    </div>
  </div>
</div>
`;

exports[`containers/Profile/components/.../AwardItem when editMode is true renders correctly 1`] = `
<div>
  <div
    className="profile-experience-block profile-unverified-status"
  >
    <div
      className="profile-experience-btns"
    >
      <button
        className="edit-btn"
        onClick={[Function]}
      >
         
      </button>
    </div>
    <div>
      <div
        className="relative"
      >
        <div
          className="Avatar Avatar-white Avatar-size-109"
        >
          <img
            alt="logo"
            src="https://appii.s3.eu-west-2.amazonaws.com/organisation_logos/aa-plc1509336690734.jpg"
          />
        </div>
        <p
          className="visible-xs visible-sm"
        >
          <i
            className="profile-status-ico tooltip-position-top tooltip-movable tooltip-dark"
            data-tooltip="Not verified"
          >
             
          </i>
        </p>
      </div>
      <div
        className="profile-experience-info"
      >
        <i
          className="profile-status-ico tooltip-position-top tooltip-movable tooltip-dark hidden-xs hidden-sm"
          data-tooltip="Not verified"
        >
           
        </i>
        <p>
          Award name
        </p>
        <span>
          AA plc
        </span>
        <em>
          
        </em>
      </div>
    </div>
  </div>
</div>
`;
